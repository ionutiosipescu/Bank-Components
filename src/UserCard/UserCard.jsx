import React from "react";
import { Icon } from "../ProfileComponent/ProfileComponent.style";
import {
  UserCardBody,
  UserCardContainer,
  UserCardHeader,
  UserCardInfo,
} from "./UserCard.style";

import { RiMailLine, RiPhoneLine } from "react-icons/ri";
import { Link } from "react-router-dom";

function UserCard({ image, name, email, phone, onClick, to }) {
  return (
    <UserCardContainer as={Link} to={to} onClick={onClick}>
      <UserCardHeader>
        <img src={image} alt="avatar" />
        <h3>{name} </h3>
      </UserCardHeader>
      <UserCardBody>
        <UserCardInfo>
          <Icon>
            <RiMailLine size={20} />
          </Icon>
          <div>
            <p>Email</p>
            <h5>{email}</h5>
          </div>
        </UserCardInfo>
        <UserCardInfo>
          <Icon>
            <RiPhoneLine size={20} />
          </Icon>
          <div>
            <p>Phone</p>
            <h5>{phone}</h5>
          </div>
        </UserCardInfo>
      </UserCardBody>
    </UserCardContainer>
  );
}

export default UserCard;
