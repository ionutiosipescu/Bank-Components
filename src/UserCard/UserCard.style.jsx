import styled from "styled-components";

export const UserCardContainer = styled.div`
  width: calc(25% - 20px);
  height: calc(50% - 20px);
  padding: 10px 30px;
  margin: 10px;
  border-radius: 20px;
  box-shadow: 0px 0px 20px -10px var(--gray-dark);
  color: var(--text);
  background-color: var(--white);
  cursor: pointer;
  transition: 0.3s;
  &:hover {
    scale: 105%;
    transition: 0.3s;
  }
  & img {
    aspect-ratio: 1/1;
    object-fit: cover;
  }
`;

export const UserCardHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 45%;
  & > img {
    width: 30%;
    border-radius: 50%;
  }
`;

export const UserCardBody = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 100%;
  height: 55%;
`;

export const UserCardInfo = styled.div`
  display: flex;
  width: 100%;
  & > div:last-of-type {
    margin-left: 20px;
  }
`;
