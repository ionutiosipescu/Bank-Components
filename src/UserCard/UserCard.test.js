import React from "react";
import { shallow, mount, render } from "enzyme";
import UserCard from "./UserCard";

it("expect to render UserCard component", () => {
  expect(shallow(<UserCard />)).toMatchSnapshot();
});
