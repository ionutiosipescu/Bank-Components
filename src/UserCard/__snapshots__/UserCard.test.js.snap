// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render UserCard component 1`] = `
<styled.div
  as={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
>
  <styled.div>
    <img
      alt="avatar"
    />
    <h3>
       
    </h3>
  </styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <RiMailLine
          size={20}
        />
      </styled.div>
      <div>
        <p>
          Email
        </p>
        <h5 />
      </div>
    </styled.div>
    <styled.div>
      <styled.div>
        <RiPhoneLine
          size={20}
        />
      </styled.div>
      <div>
        <p>
          Phone
        </p>
        <h5 />
      </div>
    </styled.div>
  </styled.div>
</styled.div>
`;
